beacon/handlers.go

```diff
@@ -185,7 +197,11 @@ func (pd *ProtocolDriver) addPotentiallyValidProposal(proposal []byte) {
 
 	pd.mu.Lock()
 	defer pd.mu.Unlock()
-	pd.incomingProposals.potentiallyValid = append(pd.incomingProposals.potentiallyValid, proposal)
+	if pd.current.incomingProposals.valid == nil {
```

Comment:
Did you mean `potentiallyValid`?
```suggestion
	if pd.current.incomingProposals.potentiallyValid == nil {
```

Reply:
oops. changed and added test

Reply:
https://github.com/spacemeshos/go-spacemesh/actions/runs/1756359387

i am surprised all the system tests passes. but then every node is run in a controlled environment.